specs/avatar.md

```diff
@@ -0,0 +1,122 @@
+# Avatar
+
+## Overview
+
+An avatar is a graphical represention of a user or object.
+
+### Use Cases
+
+A common use case would be to display an image or text (usually initials) of a user or an object, such as in a user profile.
+
+### Features
+- A URL for an image can be passed to the component to be displayed in the coin
+  - When no image is provided, the initials of the provided name will be displayed in the coin
+- Badge slot: Able to slot in a badge component
+- `shape`, a circle or square shape can be chosen. Any border radius for square shaped coins should be determined by the users design system values or stylesheet
+- `color`, a hexadecimal color can be provided to determine the coin background color
+- When a `link` is provided an `aria-link` attribute is added
+
+### Prior Art/Examples
+
+- [Fluent UI (Persona)](https://developer.microsoft.com/en-us/fluentui#/controls/web/persona)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/avatar/definition)
+- [Lighting Design](https://www.lightningdesignsystem.com/components/avatar/)
+- [Evergreen](https://evergreen.segment.com/components/avatar)
+- [Ant Design](https://ant.design/components/avatar/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/design-system/avatar)
+---
+### API
+
+*Component Name*
+- `fast-avatar`
+
+#### Attributes
+|   Name    | Description                                                 | Type                                |
+|-----------|-------------------------------------------------------------|-------------------------------------|
+| `name`    | Accepts name string for coin display                        | `string`                            |
+| `img-src` | Accepts URL string of image to be displayed                 | `string`                            |
+| `alt-text`| Accepts alt text for image                                  | `string`                            |
+| `link`    | Accepts a URL for the anchor source                         | `string`                            |
+| `shape`   | Determines the avatar coin shape. Default will be a circle. | `string: default | circle | square` |
+| `fill`    | Sets the background fill of the avatar coin.                | `string: hex color`                 |
+| `color`   | Sets the color of the avatar coin text.                     | `string: hex color`                 |
+
+### Anatomy and Appearance
+
+*Template*
+```
+${when(x => x.link, html`
+  <template 
+    role="aria-link">
+    <div class="avatar-coin">
+      <a class="avatar-link" href="${ x => x.link}">
+        ${when(x => x.imgSrc, html`
+          <img 
+            class="avatar-image" 
+            alt="${ x => x.alt }" 
+            src="${ x => x.imgSrc }" />
+        `)}
+        <span class="avatar-name">${x => x.name}</span>
+      </a>
```

Comment:
Late to the party here, and this is 100% dependent on implementation -

In this case, I would expect we may need to manage tabindex as well as click and keypress events to mimic the native anchor element which is nested within here. An alternative to this could be to delegate focus to the anchor within. In that case, I would not expect the `aria-link` role as the host role will be ignored and focus will defer to the internal anchor. If we went that route, we could also consider using a single template and just not passing href if it's not provided. While that would still have an anchor in the shadow, it wouldn't receive focus and should be treated as text by AT.

Not an imperative that we build to the above, but something to consider which may simplify the control and implementation and allow us to offload native behavior to the native elements.

Reply:
Thanks for the feedback!

This is good. VoiceOver does focus on the `a` tag itself when navigating to the component. The `alt` attribute is also read. I'll play around with some of the approaches you mentioned, my goal is to make the implementation as accessible as possible without over complicating the template.

Reply:
I pushed some changes up that I think might be better. This will have `delegatesFocus` on. A little more of a simplified approach. The only thing is if no link is provided then there will be an empty `<a>` wrapping the `img` and `span`, which I've read is [okay](https://www.w3.org/TR/2011/WD-html5-20110525/text-level-semantics.html#the-a-element). Any thoughts, @EisenbergEffect, @chrisdholt?

Reply:
Sweet - yeah, that should be ok. AT should treat an anchor without an href like its contents. If it's text, such as below, it would be like text:
```
<a>This is text</a>
<a href="#">This is a link</a>
```